UPGRADING.md

```diff
@@ -4,11 +4,27 @@ Make sure that you're running the latest Sprockets 3 release. This document is a
 
 This upgrading guide touches on:
 
+- Upgrading as a Rails dependency
 - Source Maps
 - Manifest.js
+  - In a Rails app, possible backwards incompatible changes in how top-level targets are determined.
 - ES6 support
 - Deprecated processor interface in 3.x is removed in 4.x
 
+## Upgrading as a Rails Dependency
```

Comment:
I don't think this belongs here. This is more documentation how bundler and rubygems works than it is about upgrading sprockets. Also sass-rails is not the only gem that can prevent you from upgrading.

Reply:
It is _literally_ about upgrading sprockets, and while many things _can_ prevent you from upgrading, this is the only included in a stock `rails new-created Gemfile  which _does_, and I know from talking to people it has confused real people -- the fact that Rails 6.0 (accidentally) generates a Gemfile that will not allow sprockets 4.0 is particularly confusing.  And people were getting confused about why some projects were getting _unintentional_ sprockets 4 updates with `bundle update` (or CI on gem source without a Gemfile.lock) which in some cases broke their build, while other projects were not. 

So if we have evidence it actually is confusing people around upgrading, why not explain it in the place people would go to look when they are confused about upgrading?

But in the end its sprockets maintainers call, I just wanted to make one more attempt to advocate for it!